tests/components/jewish_calendar/test_binary_sensor.py

```diff
@@ -1,301 +1,158 @@
     ),
 ]
 
-MELACHA_TEST_IDS = [
-    "currently_first_shabbat",
-    "after_first_shabbat",
-    "friday_upcoming_shabbat",
-    "upcoming_rosh_hashana",
-    "currently_rosh_hashana",
-    "second_day_rosh_hashana_night",
-    "second_day_rosh_hashana_day",
-    "currently_shabbat_chol_hamoed",
-    "upcoming_two_day_yomtov_in_diaspora",
-    "currently_first_day_of_two_day_yomtov_in_diaspora",
-    "currently_second_day_of_two_day_yomtov_in_diaspora",
-    "upcoming_one_day_yom_tov_in_israel",
-    "currently_one_day_yom_tov_in_israel",
-    "after_one_day_yom_tov_in_israel",
-]
-
 
 @pytest.mark.parametrize(
-    (
-        "now",
-        "candle_lighting",
-        "havdalah",
-        "diaspora",
-        "tzname",
-        "latitude",
-        "longitude",
-        "result",
-    ),
-    MELACHA_PARAMS,
-    ids=MELACHA_TEST_IDS,
+    ("location_data", "test_time", "results"), MELACHA_PARAMS, indirect=True
 )
 async def test_issur_melacha_sensor(
     hass: HomeAssistant,
-    now,
-    candle_lighting,
-    havdalah,
-    diaspora,
-    tzname,
-    latitude,
-    longitude,
-    result,
+    test_time: dt,
+    results: dict[str, Any],
+    config_entry: MockConfigEntry,
 ) -> None:
     """Test Issur Melacha sensor output."""
-    time_zone = dt_util.get_time_zone(tzname)
-    test_time = now.replace(tzinfo=time_zone)
-
-    await hass.config.async_set_time_zone(tzname)
-    hass.config.latitude = latitude
-    hass.config.longitude = longitude
-
+    sensor_id = "binary_sensor.jewish_calendar_issur_melacha_in_effect"
     with freeze_time(test_time):
-        entry = MockConfigEntry(
-            title=DEFAULT_NAME,
-            domain=DOMAIN,
-            data={
-                CONF_LANGUAGE: "english",
-                CONF_DIASPORA: diaspora,
-                CONF_CANDLE_LIGHT_MINUTES: candle_lighting,
-                CONF_HAVDALAH_OFFSET_MINUTES: havdalah,
-            },
-        )
-        entry.add_to_hass(hass)
-        await hass.config_entries.async_setup(entry.entry_id)
+        config_entry.add_to_hass(hass)
+        await hass.config_entries.async_setup(config_entry.entry_id)
         await hass.async_block_till_done()
+        assert hass.states.get(sensor_id).state == results["state"]
 
-        assert (
-            hass.states.get(
-                "binary_sensor.jewish_calendar_issur_melacha_in_effect"
-            ).state
-            == result["state"]
-        )
-
-        with freeze_time(result["update"]):
-            async_fire_time_changed(hass, result["update"])
-            await hass.async_block_till_done()
-            assert (
-                hass.states.get(
-                    "binary_sensor.jewish_calendar_issur_melacha_in_effect"
-                ).state
-                == result["new_state"]
-            )
+    with freeze_time(results["update"]):
+        async_fire_time_changed(hass, results["update"])
+        await hass.async_block_till_done()
```

Comment:
why do we freeze time like this rather than with the freezer? (and use `freezer.move_to` and `freezer.tick`)?

Reply:
I wasn't aware of it :P